homeassistant/components/stream/worker.py

```diff
@@ -0,0 +1,142 @@
+"""Proides the worker thread needed for processing streams."""
+import asyncio
+from fractions import Fraction
+import io
+
+from .core import Segment, StreamBuffer
+
+
+def generate_audio_frame():
+    """Generate a blank audio frame."""
+    from av import AudioFrame
+    audio_frame = AudioFrame(format='dbl', layout='mono', samples=1024)
+    audio_bytes = b''.join(b'\x00\x00\x00\x00\x00\x00\x00\x00'
+                           for i in range(0, 1024))
+    audio_frame.planes[0].update(audio_bytes)
+    audio_frame.sample_rate = 44100
+    audio_frame.time_base = Fraction(1, 44100)
+    return audio_frame
+
+
+def create_stream_buffer(stream_output, video_stream, audio_frame):
+    """Create a new StreamBuffer."""
+    import av
+    a_packet = None
+    segment = io.BytesIO()
+    output = av.open(
+        segment, mode='w', format=stream_output.format)
+    vstream = output.add_stream(
+        stream_output.video_codec, video_stream.rate)
+    # Fix format
+    vstream.codec_context.format = \
+        video_stream.codec_context.format
+    # Check if audio is requested
+    astream = None
+    if stream_output.audio_codec:
+        astream = output.add_stream(
+            stream_output.audio_codec, 44100)
```

Comment:
Shall 44k be configurable or defined as a CONST

Reply:
Made a constant